server/aws-lsp-partiql/src/server/syntax-highlighting/parser-tokens.ts

```diff
@@ -0,0 +1,210 @@
+import Parser from 'web-tree-sitter'
+import parserBase64 from '../tree-sitter-parser/tree-sitter-parser-inline'
+import wasmBinaryArray from '../tree-sitter-parser/tree-sitter-inline'
+import {
+    SemanticTokenTypes,
+    SemanticTokens,
+    SemanticTokenModifiers,
+    Range,
+} from '@aws/language-server-runtimes/server-interface'
+import { SemanticTokensBuilder } from 'vscode-languageserver/node'
+import { semanticTokensLegend } from '../language-service'
+
+export interface SemanticToken {
+    range: Range
+    tokenType: SemanticTokenTypes
+    tokenLength: number
+    /**
+     * An optional array of modifiers for this token
+     */
+    tokenModifiers?: SemanticTokenModifiers[]
+}
+
+export const string2TokenTypes: { [key: string]: SemanticTokenTypes } = {
+    tuple_punc_start: SemanticTokenTypes.operator,
+    tuple_punc_end: SemanticTokenTypes.operator,
+    tuple_punc_separator: SemanticTokenTypes.operator,
+    pair_punc_separator: SemanticTokenTypes.operator,
+    array_punc_start: SemanticTokenTypes.operator,
+    array_punc_end: SemanticTokenTypes.operator,
+    array_punc_separator: SemanticTokenTypes.operator,
+    bag_punc_start: SemanticTokenTypes.operator,
+    bag_punc_end: SemanticTokenTypes.operator,
+    bag_punc_separator: SemanticTokenTypes.operator,
+    ion: SemanticTokenTypes.string,
+}
+
+// Global or service-level variables
+let parserInitialized = false
+let globalParser: Parser | null = null
+
+async function ensureParserInitialized() {
+    if (!parserInitialized) {
+        globalParser = await initParser()
+        parserInitialized = true
+    }
+}
+
+// Initialize and prepare the parser
+async function initParser() {
```

Comment:
This code exists twice now right? here and in parserUtil.ts, or is there a difference? (I do think having this scaffolding code in a separate util file is a good idea)

Reply:
Yeah, I placed the initialization in a separate utility because hover help and signature help will also use it later. It was an oversight on my part not to sync the code in `parser-tokens.ts` during the modification. Thanks for pointing it out!